src/polytool.ts

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+import * as commandLineCommands from 'command-line-commands';
+import {HelpCommand} from './commands/help';
+import {ServeCommand} from './commands/serve';
+import {Command} from './commands/command';
+
+export class Polytool {
+
+  commandDescriptors = [];
+  commands : Map<String, Command> = new Map();
+  cli : commandLineCommands.CLI;
+
+  constructor() {
+    this.addCommand(new HelpCommand(this.commands));
```

Comment:
Nit: I'd rather see this in bin/, or a static method that provides a polytool configured with a default set of commands. It feel better to me for programmatic consumers to be able to add commands piecemeal.